File: python/55-Jump-Game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        goal = len(nums) -1
        
        for i in range(len(nums) - 2, -1, -1):
            if i + nums[i] >= goal:
                goal = i
        return goal == 0

File: swift/55-Jump-Game.swift
class Solution {
    func canJump(_ nums: [Int]) -> Bool {
        var goal = nums.count - 1
        
        for i in stride(from: nums.count-2, through: 0, by: -1){
            if i + nums[i] >= goal {
                goal = i
            }
        }
        return goal == 0
    }
}